prime-router/src/test/kotlin/azure/SettingFacadeTests.kt

```diff
@@ -208,4 +216,12 @@ class SettingFacadeTests {
         assertThat(receiver?.name).isEqualTo("elr-test")
         assertThat(receiver?.format).isEqualTo(Report.Format.CSV)
     }
+
+    @Test
+    fun `get receiver test failure with invalid name`() {
+        setupReceiverDatabaseAccess()
+        // verifies solve for #4002 - an overly-long exception name should not be throwing an exception, instead failing as normal
+        val receiver = SettingsFacade(testMetadata(), accessSpy).findReceiver("ignore.ignore-waters.ignore-waters")
```

Comment:
Incomplete unit test.  You need to test both good and bad inputs.  What happens if the service is not specified, but the org is (e.g. "ignore" only)?